src/wireguardif.h

```diff
@@ -141,6 +144,11 @@ err_t wireguardif_peer_is_up(struct netif *netif, uint8_t* pubkey);
 
 // Updates config of an added peer
 err_t wireguardif_update_peer(struct netif *netif, struct wireguardif_peer *peer, uint8_t* public_key);
+
+// Recieves packet from network
+void wireguardif_network_rx(void *arg, struct udp_pcb *pcb, struct pbuf *p, const ip_addr_t *addr, u16_t port);
+
+conn_state_t wireguardif_get_derp_conn_status(struct netif *netif);
```

Comment:
So a quick question, I see that `wireguardif_get_derp_conn_status()` is exposed, but not `...set_derp_endpoint()`. From API we will receive a list of online relay servers, and thus we will have to select the most suitable one to talk to. The logic for selecting it may be implemented separately or within `derp.c/derp.h` both variants IMHO are ok, but if we would like to implement server selection outside of derp.c/derp.h, then we will likely need to expose both, `get_derp_conn_status()` as well as `...set_derp_endpoint()`.

On the other hand - if we would like to implement relay server selection within derp.c/derp.h -> then the interface should be able to pass full list of derp servers to it. 

I think currently we are not exposing enough of the interface for first variant or exposing a bit different (e.g. single server instead of list of servers) for the second variant.

Is that the case, or maybe I am missing some sort of third variant?

Reply:
So my understanding had been the second variant where the derp selection logic sits outside of `derp.h/c` and a single server is passed to it. Atm `esp_wireguard.h` does expose `esp_wireguard_set_derp_server ` so does `derp.h` as `set_derp_endpoint ` but maybe the layering could be better, as in, `esp_wireguard` -> `wireguardif` -> `derp` instead of `esp_wireguard` -> `derp` which is the current implementation.

Reply:
Oh, I have missed the `esp_wireguard.h` part. Which is all good, then the interface makes sense.